selectors/example-selectors.md

```diff
@@ -180,39 +204,19 @@ Selector:
     sequence:
       ExploreFields:
         fields:
-          "Links"
+          "Name":
+            Matcher:
```

Comment:
So these additional matcher leafs are kinda interesting, and I wonder if we shouldn't add some more comment on them?  And also this has some interaction with Rod's comments about how much we want to blur block boundaries in the example data...

If one is just worried about "retrieve a full sub-DAG", as you set it up in the first line of the section... well, "retrieve" sort of invokes concepts that are halfway to something higher, like graphsync.  "Retrieve" is a concept entangled with _blocks_.

In practice, one will "retrieve" a bunch more _nodes_ than the ones explicitly matched or traversed searching for matches, because they'll ride along _in the block_ of anything that was explicitly matched or traversed.

Probably we want to be careful to use words about "match"/"visit"/"yield"(?) (I haven't been completely consistent on these to date -- Mea Cupla, and <3 help trying to improve this) most of the time.  Or we could introduce these details about blocks-vs-nodes and those impliciations during these examples too, but I think maybe that might be a little much (and if we do that, maybe we should do it in a non-unixfs example just to reduce the unrelated potential cognitive load for readers...).